### DISCUSSION TITLE
acr_values and claims object inside authorization request token

### DISCUSSION THREAD

**Discussion by lucamontano98** [OpenIDC/mod_auth_openidc#1143] [Q&A]
Good morning,
can you tell me how can I include acr_values and claims object inside authorization request token? Thanks.

Luca

**Comment by zandbelt** [comment]
I'm not entirely sure what "authorization request token" refers to but perhaps this helps: https://github.com/OpenIDC/mod_auth_openidc/wiki#13-how-can-i-add-custom-parameters-to-the-authorization-request

**Reply by lucamontano98** [reply]
Sorry, with "authorization request token" I'm linking to the signed request object which Relying Party sends to the authorization endpoint of the OpenID Provider. Now there is the "acr_values" in the request object, but I don't have the "claims" field inside the same object ("request"). I use the authorization code flow. Can you tell me how can I have the missing field "claims" in the request object using the configuration file? Thanks a lot.

**Reply by zandbelt** [reply]
you can add the claims field to the authorization request parameters and copy them into the request object using the `copy_from_request` or `copy_and_remove_from_request` primitives in the `OIDCRequestObject` value, or add them statically using the `static` primitive as done in the example, see: https://github.com/OpenIDC/mod_auth_openidc/blob/v2.4.14.4/auth_openidc.conf#L975-L979
